streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -624,9 +624,11 @@ StandbyTask standbyTask(final TopicPartition partition) {
         return null;
     }
 
-    // TODO K9113: this is used from StreamThread only for a hack to collect metrics from the record collectors inside of StreamTasks
-    // Instead, we should register and record the metrics properly inside of the record collector.
-    Map<TaskId, StreamTask> fixmeStreamTasks() {
-        return tasks.values().stream().filter(t -> t instanceof StreamTask).map(t -> (StreamTask) t).collect(Collectors.toMap(Task::id, t -> t));
+    Map<MetricName, Metric> producerMetrics() {
```

Comment:
nit: These two functions are not for testing only.

Reply:
Thanks for the feedback; I didn't understand this particular comment, though.

Reply:
Ah, I just found what you were talking about:
```
    // below are for testing only
```

I didn't notice that up there.